### ISSUE TITLE
Create a deploy command

### ISSUE THREAD

**Issue by anavarre** [acquia/cli#201]
In Acquia CLI, there's no equivalent to `ads deploy`. This means in Cloud IDE users might be confused as to how they can deploy to any Cloud environment.

Here's how `ads deploy` currently works:

```
$ ads deploy
Select corresponding Acquia Cloud Application: mydrupalwebsite
? Select source:  Stage
? Select target environment:  Prod
? Are you sure you want to deploy to Prod? Yes
? Do you want to back up the databases before deploying the code? No
? Do you want to clear Varnish cache after the deploy? Yes

Summary:

 ›   Warning: The databases from Prod will not be backed up before deployment.
The environment Prod will be switched to progress.
Varnish cache will be cleared from Prod after deployment.

? Continue? Yes
  ✔ mydrupalwebsite: deploying progress to Prod
  ✔ Clearing Varnish cache

Successfully deployed progress to Prod
? Open the website? No
```

Perhaps initially in an IDE there could be a wrapper to `ads deploy` until we have a full-blown command.

**Comment by grasmash** [comment]
I’m not convinced that this is valuable, and I think that it may be outside of a defined scope for this tool.

What are we trying to accomplish? I would expect that deployment would involve the following steps:

* git add, commit, and push
* check out tag or branch on a given cloud environment
* run post deploy commands like cache clear or data as updates

I’m not sure that any of those things make sense as the responsibility of Acquia CLI.

**Comment by anavarre** [comment]
Real-life use case from an Acquia CLI user in IDEs:

- Pulled code, DB, files in the IDE
- Replaced codebase by fresh new install
- User blocked on uploading DB and files back to the Cloud Platform
- Got back to Dev Studio CLI to be unblocked

**Comment by grasmash** [comment]
OK, it sounds like you’re envisioning a different scope for this command. Are you suggesting that the command should push files and database upstream? Can you write the user story and requirements?

**Comment by aadityajshah** [comment]
I think `ads deploy` or a similar feature is definitely needed. For example, my current `dev` environment is running an older version of Drupal 8. In my CloudIDE I have created a new multi-site running Drupal 9. I should be able to push out the code to the repo, as well as update the DB as needed in this instance. Also, it should work without BLT as well 

I agree that it should not necessarily be an ACLI command, but it should allow for a `deploy` of some sort to be implemented through the CloudIDE.

**Comment by grasmash** [comment]
For the sake of clarity, I think it might be better to do something like this:

* Rename ‘refresh’ command to ‘pull’ 
* Add ‘pull:files’, ‘pull:code’ and ‘pull:database’ and ‘pull: run-scripts’ commands
* Add ‘push:database’ command
* Add ‘push:files’ command 

If we add a ‘push:code’ command, I would prefer it to be non-functional and for it to simply emit a message like “Please use git to push code. If you’re using composer, ensure that you commit all required vendor dependencies to your hosting environment.”

I don’t wanna be involved in the process of building an artifact at all. That is what BLT currently does, and it something the cloud should eventually do.

**Comment by ba66e77** [comment]
This is especially important since the drush setup on Cloud IDE does not allow you push a database with something like `drush sql-sync @self @remote.dev`. As a result there's no apparent way to get a database from a Cloud IDE instance to a Cloud environment.

**Comment by grasmash** [comment]
Yes, consider this on the roadmap.

**Comment by grasmash** [comment]
Resolved by #305.
